tests/html/bluetooth/bluetooth_device_disconnect.html

```diff
@@ -9,85 +9,82 @@
     <button type="button" onclick="onReconnectButtonClick()">Reconnect()</button>
     <pre id="log"></pre>
+    <script src="bluetooth_functions.js"></script>
     <script>
     var bluetoothDevice;
 
     function onScanButtonClick() {
         clear();
-        var options = {filters: []};
+        var options = {filters: [{}]};
 
         var filterService = document.getElementById('service').value;
-        if (filterService.startsWith('0x'))
-            filterService = parseInt(filterService, 16);
-
-        if (filterService)
-            options.filters.push({services: [filterService]});
+        if (filterService) {
+            if (filterService.startsWith('0x'))
+                filterService = parseInt(filterService, 16);
+            options.filters[0].services = [filterService];
+        }
 
         var filterName = document.getElementById('name').value;
         if (filterName)
-            options.filters.push({name: filterName});
+            options.filters[0].name = filterName;
 
         var filterNamePrefix = document.getElementById('namePrefix').value;
         if (filterNamePrefix)
-            options.filters.push({namePrefix: filterNamePrefix});
-
-        bluetoothDevice = null;
+            options.filters[0].namePrefix = filterNamePrefix;
 
         try {
-            log('Requesting Bluetooth Device...');
+            clear();
+            log(timeStamp() + 'Requesting Bluetooth Device...');
             bluetoothDevice = window.navigator.bluetooth.requestDevice(options);
+            log(timeStamp() + 'Connecting to Bluetooth Device...');
             connect();
         } catch(err) {
-            log(err);
+            log(timeStamp() + err);
         }
     }
 
     function onDisconnectButtonClick() {
         clear();
         if (!bluetoothDevice)
-            return;
+            return log(timeStamp() + '> There is no connected Bluetooth Device instance, from which we can disconnect');
 
         try {
-            log('Disconnecting from Bluetooth Device...');
+            log(timeStamp() + 'Disconnecting from Bluetooth Device...');
             if (bluetoothDevice.gatt.connected) {
                 bluetoothDevice.gatt.disconnect();
-                log('> Bluetooth Device connected: ' + bluetoothDevice.gatt.connected);
+                log(timeStamp() + '> Bluetooth Device connected: ' + bluetoothDevice.gatt.connected);
             } else {
-                log('> Bluetooth Device is already disconnected');
+                log(timeStamp() + '> Bluetooth Device is already disconnected');
             }
         } catch(err) {
-            log(err);
+            log(timeStamp() + err);
         }
     }
 
     function onReconnectButtonClick() {
         clear();
         if (!bluetoothDevice)
-            log('> There is no connected Bluetooth Device instance')
+            log(timeStamp() + '> There is no connected Bluetooth Device instance, for which we can reconnect')
```

Comment:
http://english.stackexchange.com/questions/90130/is-this-usage-of-for-which-correct